url4e79c36700_JacenLakiir_Tivoo_tgz-pJ8-MainJ8/src/tivoo/output/HTMLBuilder.java

```diff
@@ -0,0 +1,382 @@
+        {
+            String eventDay = formatDay(currentEvent);
+            if (!sortedEvents.containsKey(eventDay))
+                sortedEvents.put(eventDay, new ArrayList<Event>());
+            sortedEvents.get(eventDay).add(currentEvent);
+        }
+        return sortedEvents;
+    }
+    
+    protected Map<String, List<Event>> sortByDate (List<Event> eventList)
+    {
+        Map<String, List<Event>> sortedEvents = new TreeMap<String, List<Event>>();
+        for (Event currentEvent : eventList)
+        {
+            String eventDay = formatShortDate(currentEvent);
+            if (!sortedEvents.containsKey(eventDay))
+                sortedEvents.put(eventDay, new ArrayList<Event>());
+            sortedEvents.get(eventDay).add(currentEvent);
+        }
+        return sortedEvents;
+    }
+    
+    protected String getTitle ()
+    {
+        return TITLE;
+    }
+    
+    protected String getUniqueCSS ()
+    {
+        return UNIQUE_CSS;
+    }
+    
+    private static List<String> initializeDaysList ()
+    {
+        String[] days =
+            new String[] {
+                    "Sunday",
+                    "Monday",
+                    "Tuesday",
+                    "Wednesday",
+                    "Thursday",
+                    "Friday",
+                    "Saturday" };
+        List<String> dayList = new ArrayList<String>();
+        for (String d : days)
+            dayList.add(d);
+        return dayList;
+    }
+    
+    private static List<String> initializeMonthsList ()
+    {
+        String[] months =
+            new String[] {
+                    "January",
+                    "February",
+                    "March",
+                    "April",
+                    "May",
+                    "June",
+                    "July",
+                    "August",
+                    "September",
+                    "October",
+                    "November",
+                    "December" };
+        List<String> monthList = new ArrayList<String>();
+        for (String m : months)
+            monthList.add(m);
+        return monthList;
+    }
+
+    private String determineDetailPageDirectory (String summaryPageFileName)
+    {
+        StringBuilder detailPageDirectory = new StringBuilder();
+        if (summaryPageFileName.contains("/"))
+        {
+            int lastFolder = summaryPageFileName.lastIndexOf("/");
+            detailPageDirectory.append(summaryPageFileName.substring(0, lastFolder + 1));
+            detailPageDirectory.append(DETAIL_PAGE_FOLDER);
+        }
+        else
+        {
+            detailPageDirectory.append(DETAIL_PAGE_FOLDER);
+        }
+        return detailPageDirectory.toString();
+    }
+    
+    private void buildSummaryPage (List<Event> eventList) throws IOException
+    {
+        if (mySummaryPageFileName.contains("/"))
+        {
+            File outputDirectory = new File(myDetailPageDirectory);
+            outputDirectory.mkdirs();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Confirm that `mkdirs` was successful. Handle any failures that happened. If you fail silently, then you might encounter errors that are harder to debug. See the following diffs on GitHub that make this code change: [diff1](https://github.com/qunarcorp/qmq/commit/12f1d9d6a57a02bc2415cb2af9f4388a77b12c02#diff-6b67be6e17f24fdeddb7c725ac8ec5600d6e63be0bdca0963a83d530e6d55e35L36), [diff2](https://github.com/apache/jmeter/commit/1ec61206779b3f84df18516682b4791bf015178f#diff-cd74304763c097fbc085ac2bc952b46ab3843ac825cacacad89e8064296e955aL330)

Learn more: [File.mkdirs in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#mkdirs--)